### PR TITLE
Update GitHub documentation

### PR DESCRIPTION
**Pull request by samglover**
This is a PR for updates related to the LIT Lab's 2024 summer cohort. In the course of that work it became clear we needed to update the GitHub instructions to help student teams work together. This update primarily addresses that, along with some additional minor changes.

Some notes:
* I removed a whole lot of stuff from the GitHub page and reorganized it around workflow, how-tos, best practices, and troubleshooting. I know how it can feel to get edited this extensively, and I hope @plocket, who wrote most or all of the existing page, will forgive me.
* A lot of what I removed (as opposed to reorganized and revised) was (1) duplicative of GitHub or Docassemble documentation, so I replaced it with links, and (2) related to managing files, packages, and projects in the Docassemble playground but not necessarily related to GitHub (and a lot of this can also be found in the Docassemble docs).
* I resolved the broken links and anchors related to my changes. I did not resolve broken anchors related to the ALKiln pages since as far as I know they are still a work in progress.
* I will continue working on this branch after this PR is merged. There's lots more to do but I felt this was worth pushing out now.

### CODE REVIEW

File: docs/github.md
Comment (nonprofittechy) [inline docs/github.md:176]: I think we still need to document this step as it's the first step after using the Weaver. It could link to another part of the documentation though

**Comment by samglover** [comment]
@nonprofittechy I made a few updates in order to include information about uploading a package, and that led me to add some further notes on naming packages before publishing to GitHub (addressing #434). Which then made me realize the name formats page was not showing up anywhere in the sidebar, so I fixed that, too.

**Review by nonprofittechy — APPROVED** [review]

**Review by plocket — APPROVED** [review]
No worries! It was a group effort and a lot of fun. The audience was very different and some external documentation didn't exist, or was insufficient in our opinion, when we first created this, so there is plenty that's redundant. Anyway, documentation gets outdated.

I love the work here, though I miss some of the pictures. It's wonderfully concise and to-the-point. I like leaving the docs up to the people who make the software (where possible, of course). I made some suggestions, but you can take them or leave them. A couple are just plain language tweaks, and I later decided this probably isn't the PR for deep plain language revisions and I'm not the expert anyway. There are some typo fixes too.

I think it's basically ready for prime time. I'll approve and if you want me to read updates just let me know 👍

File: docs/github.md
Comment (samglover) [reply docs/github.md:78]: I agree, but I also think it makes sense to see this in the sidebar:

> **How To**
> * Commit Your Code
> * Create a Pull Request
> * Resolve Conflicts
> * Review Pull Requests

And the section on reviewing pull requests is for reviewers, not for creating it in the first place. So I think I'm going to leave it for now, but I will keep it in mind for future updates.

I'm sure the next cohort of interview builders will run into different problems that I can use as prompts for future updates.

File: docs/github.md
Comment (samglover) [reply docs/github.md:82]: I don't think I know enough about how merge conflicts change the head branch before the merge, but I'd like to. Can you give me the short version, perhaps in Teams?

File: docs/github.md
Comment (samglover) [reply docs/github.md:109]:
```suggestion
Then, delete the playground project you created for the review.
```

File: docs/github.md
Comment (samglover) [reply docs/github.md:32]:
```suggestion
1. Create a new project in your Docassemble playground. Give it a name that describes the work to be done, then pull the `main` branch of your interview from GitHub into the project.
2. When you make your first commit from the new project, [commit to a new branch](#commit-your-code).
3. When you finish working on the branch, commit it one last time, then go to the repository in GitHub and [create a pull request](#create-a-pull-request). If you are working with someone on the interview, request a review from them.
4. Once you have resolved any conflicts and any reviewers have signed off, merge the pull request.
5. Decide on the next task or issue you want to work on and start over at #1.

:::warning
When you pull a GitHub repository to a playground project, the files in the repository will overwrite any files with the same name in your project **without warning**. Create a new project, first, if you don't want to risk this.
:::
```

File: docs/github.md
Comment (samglover) [reply docs/github.md:495]: Good catch. There isn't an obvious place for it here, so until I have a chance to write more about pulling from GitHub I added this warning to the Workflow section so it won't get missed.

File: docs/github.md
Comment (samglover) [reply docs/github.md:145]:
```suggestion
As a general rule, avoid editing the branch called `main`, the default branch. Only update `main` with pull requests from other branches. As another general rule, avoid creating branches from branches other than `main`—this is a recipe for merge conflicts.
```

File: docs/github.md
Comment (samglover) [reply docs/github.md:151]:
```suggestion
* Branch names are usually short. They are a description of the purpose of the changes you will make—a description that you or your teammates can understand at a glance.
* While you are working on a branch, [commit early and often](#commit-early--often), and use the commit message to describe the change you just made to the interview code.
* Pull request titles should be a summary of the purpose of your changes. They are usually more detailed than the branch name. Pull request descriptions should provide context, [mention related issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#about-linked-issues-and-pull-requests), and generally explain the reason for the change.
```

File: docs/github.md
Comment (nonprofittechy) [reply docs/github.md:82]: I think what Michelle means is that when you use the merge editor on GitHub, it merges `main` into your feature branch. The pull request metaphor does break down a little when you do a manual edit to resolve a conflict. That is a bit surprising compared to how other edits work in a branch in GitHub, but it's a feature of Git--"pull requests" are just a fancy GitHub-ism for "merge main", and aren't a concept that Git has natively. 

Personally I think this is too nuanced for this documentation and isn't specific enough to Docassemble to go in our documentation. Unless I'm misunderstanding what Michelle is discussing.

**Pull request merged**
